backend/functions/views.py

```diff
@@ -1,30 +1,141 @@
+        output_dataset_instance_id (int): ID of the output dataset instance.
+    """
+
+    # Iterate through the languages
+    for output_language in languages:
+
+        # Convert the input_sentences list into a dataframe
+        input_sentences_df = pd.DataFrame(
+            input_sentences,
+            columns=[
+                "sentence_text_id",
+                "corrected_text",
+                "input_language",
+                "context",
+                "quality_status",
+                "metadata",
+            ],
+        )
+
+        # Keep only the clean sentences
+        input_sentences_df = input_sentences_df[
+            (input_sentences_df["quality_status"] == "Clean")
+        ]
+
+        # Check if the dataframe is empty
+        if input_sentences_df.shape[0] == 0:
+            return "No clean sentences found. Perform project export first."
+
+        # Make a sentence list for valid sentences to be translated
+        sentences_to_be_translated = input_sentences_df["corrected_text"].tolist()
+
+        # Get the translation using the Indictrans NMT API
+        translations_output = get_batch_translations_using_indictrans_nmt_api(
+            sentences_to_be_translated,
+            input_sentences_df["input_language"].iloc[0],
+            output_language,
+        )
+
+        # Check if translation output didn't return an error
+        if isinstance(translations_output, Exception):
+            return translations_output
+
+        # Collect the translated sentences
+        translated_sentences = [
+            translation["target"] for translation in translations_output
+        ]
+
+        # Check if the translated sentences are equal to the input sentences
+        if len(translated_sentences) != len(sentences_to_be_translated):
+            return "The number of translated sentences does not match the number of input sentences."
+
+        # Add the translated sentences to the dataframe
+        input_sentences_df["translated_text"] = translated_sentences
+
+        # Get the output dataset instance
+        output_dataset_instance = dataset_models.DatasetInstance.objects.get(
+            instance_id=output_dataset_instance_id
+        )
+
+        # Iterate through the dataframe
+        for index, row in input_sentences_df.iterrows():
+
+            # Get the values for the TranslationPair model
+            sentence_text_id = row["sentence_text_id"]
+            input_sentence = row["corrected_text"]
+            input_language = row["input_language"]
+            translated_sentence = row["translated_text"]
+            metadata = row["metadata"]
+            context = row["context"]
+
+            # Get the output dataset instance
+            output_dataset_instance = dataset_models.DatasetInstance.objects.get(
+                instance_id=output_dataset_instance_id
+            )
```

Comment:
We already have this block outside the loop. Please remove.

Reply:
Done.